spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -470,8 +601,11 @@ case class DeltaSharingSource(
     }
     logDebug(s"start: $startOffsetOption end: $end")
 
-    val createdDf = getFileChangesAndCreateDataFrame(
-      startVersion, startIndex, isStartingVersion, endOffset
+    val createdDf = createDataFrameFromOffset(
```

Comment:
Question: It seems earlier we were calling maybeGetFileChanges as part of this, but we don't do that anymore. Is that ok?

Reply:
good catch! I've thought about it and removed. 
It's ok under the assumption that latestOffset is always called before a getBatch is called. But not ok if that assumption is not true.

It would be safer to add it. I added it back.

And there's a todo which is to see how to create tmp dir in the test so we can test the behavior of streaming checkpoint, and then we can verify if it's safe to remove it.